client/client_update.go

```diff
@@ -160,6 +161,21 @@ func processUpdateResponse(response *http.Response) (interface{}, error) {
 			return nil, errors.Wrapf(err, "failed to parse response")
 		}
 
+		if len(data.Artifact.ArtifactProvidesBuffer) > 0 {
+			// Version >= 3 includes ArtifactName in ArtifactProvides
+			// structure, which also contains ArtifactGroup.
+			var provides artifact.ArtifactProvides
+			err := json.Unmarshal(
+				data.Artifact.ArtifactProvidesBuffer, &provides)
```

Comment:
As mentioned elsewhere, I think you will need to parse into a `map[string]interface{}`, since the merged provides values are a simple key/value store.